content/en/docs/concepts/networking/route.md

```diff
@@ -0,0 +1,124 @@
+---
+title: "Route"
+linkTitle: "Route"
+weight: 3430
+description:
+---
+
+## Route
+`route` is part of `Function` resource. `route` defines how traffic from the `Gateway` listener is routed to function.
+`route` specify the `Gateway` they want to attach to using `GatewayRef`, this will allow the `route` to receive traffic from the `Gateway`.
+
+Once a synchronous `Function` is created, the function controller will:
+- If `route.gatewayRef` is not defined in function, will look for `Gateway` called `openfunction` in `openfunction` namespace, then attach to this `Gateway`.
+- If `route.hostnames` is not defined in function, will be automatically generated based on `Gateway.spec.hostTemplate`.
+- If `route.rules` is not defined in function, will be automatically generated based on `Gateway.spec.pathTemplate` or path of `/`.
+- a `HTTPRoute` custom resource will be created based on `route`. `BackendRefs` will be automatically link to the corresponding Knative service revision 
+and label `HTTPRouteLabelKey` will be added to this `HTTPRoute`.
+- Create service `{{.Name}}.{{.Namespace}}.svc.cluster.local`, this service defines an entry for the function to access within the cluster.
+- If the `Gateway` referenced by `route.gatewayRef` changed, will update the `HTTPRoute`.
+
+Once a synchronous `Function` is reconciled, the `Function` addresses and status of `route` will be shown in the `Function.status`.
+
+## Host-Based Routing
+`host-based` is the default routing mode. When `route.hostnames` are not defined in function,
+function controller will generate `route.hostnames` based on `gateway.spec.hostTemplate`. 
+If `route.rules` is also not defined, will generate `route.rules` based on path of `/`.
+
+```yaml
+apiVersion: core.openfunction.io/v1beta1
```

Comment:
All yamls to be applied should be in the form like below:
cat <<EOF | kubectl apply -f -
...
EOF